R/match_pc_gpprac.R

```diff
@@ -0,0 +1,256 @@
+#' Match postcode with GP Practice
+#'
+#' @description Match postcode with GP Practice
+#'
+#' @param data episode files
+#'
+#' @return data with matched postcode
+#' @export
+match_pc_gpprac <- function(data) {
+  # Recoding hb codes to 2018 standard ----
+  data_hb_pc <- data %>%
+    dplyr::mutate(hbtreatcode = as.character.Date(hbtreatcode)) %>%
+    dplyr::mutate(
+      dplyr::across(
+        c("hbrescode", "hbpraccode", "hbtreatcode"),
+        ~ dplyr::case_when(
+          # HB2014 to HB2018
+          .data == "S08000018" ~ "S08000029",
+          .data == "S08000027" ~ "S08000030",
+          # HB2019 to HB2018
+          .data == "S08000031" ~ "S08000021",
+          .data == "S08000032" ~ "S08000023",
+          TRUE ~ .
+        )
+      ),
+      HSCP = dplyr::case_when(
+        # HSCP2016 to HSCP2018
+        HSCP == "S37000014" ~ "S37000032",
+        HSCP == "S37000023" ~ "S37000033",
+        # HSCP2019 to HSCP2018
+        HSCP == "S37000034" ~ "S37000015",
+        HSCP == "S37000035" ~ "S37000021",
+        TRUE ~ HSCP
+      )
```

Comment:
Where has this `HSCP` variable come from? I'm not sure it's right...

Reply:
In the temp episode file no. 7 (which is the one the C09 SPSS syntax works on) there is a variable just called HSCP. Presumably this gets renamed at some point.